### PR TITLE
chore: sf undelegating / delegated cl queries

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

`/osmosis/superfluid/v1beta1/user_cl_superfluid_positions_bonded/{delegator_address}`

Returns all bonded cl positions of a user across all pools

`/osmosis/superfluid/v1beta1/user_cl_superfluid_positions_unbonding/{delegator_address}`

Returns all unbonding cl positions of a user across all pools

### CODE REVIEW

File: x/superfluid/keeper/grpc_query.go [hunk 1]
```diff
@@ -653,3 +633,73 @@ func (q Querier) UnpoolWhitelist(goCtx context.Context, req *types.QueryUnpoolWh
...
+func (q Querier) filterConcentratedPositionLocks(ctx sdk.Context, positions []model.Position, isUnbonding bool) ([]types.ConcentratedPoolUserPositionRecord, error) {
+	// Query each position ID and determine if it has a lock ID associated with it.
+	// Construct a response with the position ID, lock ID, the amount of cl shares staked, and what those shares are worth in staked osmo tokens.
+	var clPoolUserPositionRecords []types.ConcentratedPoolUserPositionRecord
+	for _, pos := range positions {
+		lockId, err := q.Keeper.clk.GetLockIdFromPositionId(ctx, pos.PositionId)
+		switch err.(type) {
+		case cltypes.PositionIdToLockNotFoundError:
+			continue
+		case nil:
```
Comment (p0mvn) [inline x/superfluid/keeper/grpc_query.go:646]: nit: is it possible to use if / else without the default case here? I don't understand why we would need the default case

File: x/superfluid/keeper/grpc_query.go [hunk 2]
```diff
@@ -653,3 +633,73 @@ func (q Querier) UnpoolWhitelist(goCtx context.Context, req *types.QueryUnpoolWh
...
+			// If we have hit this logic branch, it means that, at one point, the lockId provided existed. If we fetch it again
+			// and it doesn't exist, that means that the lock has matured.
+			lock, err := q.Keeper.lk.GetLockByID(ctx, lockId)
+			if err == errorsmod.Wrap(lockuptypes.ErrLockupNotFound, fmt.Sprintf("lock with ID %d does not exist", lock.GetID())) {
+				continue
+			}
```
Comment (p0mvn) [inline x/superfluid/keeper/grpc_query.go:652]: Can we do a comparison with `errors.Is`?

**Review by p0mvn — APPROVED** [review]

File: proto/osmosis/superfluid/query.proto [hunk 3]
```diff
@@ -125,12 +125,20 @@ service Query {
                                    "unpool_whitelist";
   }
 
-  rpc UserSuperfluidPositionsPerConcentratedPoolBreakdown(
-      UserSuperfluidPositionsPerConcentratedPoolBreakdownRequest)
-      returns (UserSuperfluidPositionsPerConcentratedPoolBreakdownResponse) {
+  rpc UserConcentratedSuperfluidPositionsBonded(
+      UserConcentratedSuperfluidPositionsBondedRequest)
```
Comment (jonator) [inline proto/osmosis/superfluid/query.proto:129]: Naming nit again: what are your thoughts on matching our staked gamm share convention of reverring to "bonded/staked" shares/positions as "delegated" (and "undelegated") as we already do?

File: proto/osmosis/superfluid/query.proto [hunk 4]
```diff
@@ -125,12 +125,20 @@ service Query {
...
+      returns (UserConcentratedSuperfluidPositionsBondedResponse) {
     option (google.api.http).get = "/osmosis/superfluid/v1beta1/"
-                                   "user_superfluid_positions_per_pool/"
-                                   "{delegator_address}/{concentrated_pool_id}";
+                                   "user_cl_superfluid_positions_bonded/"
```
Comment (jonator) [inline proto/osmosis/superfluid/query.proto:132]:
```suggestion
                                   "account_delegated_cl_positions/"
```

Matches existing queries more closely

File: proto/osmosis/superfluid/query.proto [hunk 5]
```diff
@@ -125,12 +125,20 @@ service Query {
...
+                                   "{delegator_address}";
+  }
+
+  rpc UserConcentratedSuperfluidPositionsUnbonding(
+      UserConcentratedSuperfluidPositionsUnbondingRequest)
+      returns (UserConcentratedSuperfluidPositionsUnbondingResponse) {
+    option (google.api.http).get = "/osmosis/superfluid/v1beta1/"
+                                   "user_cl_superfluid_positions_unbonding/"
```
Comment (jonator) [inline proto/osmosis/superfluid/query.proto:140]:
```suggestion
                                   "account_undelegating_cl_positions/"
```

File: x/superfluid/keeper/grpc_query.go (hunk 1 above)
Comment (czarcas7ic) [reply x/superfluid/keeper/grpc_query.go]: added here https://github.com/osmosis-labs/osmosis/pull/5691/commits/64e400454a0d14b4850ab48757dd7f7928fdf3c0

File: x/superfluid/keeper/grpc_query.go (hunk 2 above)
Comment (czarcas7ic) [reply x/superfluid/keeper/grpc_query.go]: added here https://github.com/osmosis-labs/osmosis/pull/5691/commits/64e400454a0d14b4850ab48757dd7f7928fdf3c0

File: proto/osmosis/superfluid/query.proto (hunk 3 above)
Comment (czarcas7ic) [reply proto/osmosis/superfluid/query.proto:129]: I can name delegated and undelegating, undelegated seems like the wrong name though

Comment (jonator) [reply proto/osmosis/superfluid/query.proto:129]: Yeah that's good, I was just saying it should be permutations of "delegate" instead of "bond"

Comment (czarcas7ic) [reply proto/osmosis/superfluid/query.proto]: Done here https://github.com/osmosis-labs/osmosis/pull/5691/commits/88fb86bfa7cf2518461879a8c2092ce9a831e48f

**Pull request merged**